Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.cpp

```diff
@@ -993,5 +1027,65 @@ namespace AzToolsFramework
 
             return AZStd::move(unsavedPrefabsContainer);
         }
+
+        void PrefabSaveHandler::SourceFileRemoved(
+            AZStd::string relativePath, AZStd::string scanFolder, [[maybe_unused]] AZ::Uuid sourceUUID)
+        {
+            // This gets triggered for every source file. We only need source files that are prefabs and are loaded in the current level.
+            TemplateId loadedTemplateId = s_prefabSystemComponentInterface->GetTemplateIdFromFilePath(relativePath.c_str());
```

Comment:
Can we check whether the relativePath has a ".prefab" extension or some way to early return if the source file is not a prefab file?

Reply:
I've considered doing that but the cost of fetching the extension from an AZStd::string would be higher than looking it up in a map that contains less than 100 entries (for a large level) in a map. So I'm directly looking it up in the map instead.

Reply:
OK, fair enough if we're not anticipating too many templates loaded at a time.